### ISSUE TITLE
Parameter descriptions are only outputted on referenced schema, making them invisible in docs

### ISSUE THREAD

**Issue by shroudedcode** [RobinTail/express-zod-api#929]
We recently upgraded to express-zod-api v10. This, in general, massively cleaned up our OpenAPI schema and made it work much better with code generation tools (thanks for that! 🙏). Still, it seems like it also introduced a new bug or potentially a new gotcha we weren't aware of.

Here's how to reproduce the problem:

1. Create an endpoint or middleware that has an input with a description:

```ts
export const testEndpoint = endpointFactory.build({
  input: z.object({
    cursor: z
      .string()
      .optional()
      .describe(
        'An optional cursor string used for pagination.' +
          ' This can be retrieved from the `next` property of the previous page response.',
      ),
  }),
  output: z.object({}),
  handler: async () => {
    return {}
  },
})
```

2. Generate the OpenAPI schema and look at the output:

```yml
parameters:
  - name: cursor
    in: query
    required: false
    description: GET /hris/employees parameter
    schema:
      $ref: "#/components/schemas/GetHrisEmployeesParameterCursor"
```

```yml
GetHrisEmployeesParameterCursor:
  type: string
  description: An optional cursor string used for pagination. This can be
    retrieved from the `next` property of the previous page response.
```

The custom description is outputted, but only on the referenced schema, not the actual parameter. On the parameter, the description is not only missing (which might cause documentation tools to pull the description from the schema), but specifically set to something generic (`GET /hris/employees parameter`).

The result is that no helpful information makes it all the way through to the docs:

<img width="256" alt="image" src="https://user-images.githubusercontent.com/29176678/230959250-f2430772-4a81-4a5e-ab61-0f78b15ec2ae.png">

I'm wondering: Is this expected behavior, and there's a workaround, or is this a bug introduced by the new major release? 🤔

**Comment by RobinTail** [comment]
Thank you for informing me, @shroudedcode .
Seems that it was affected by the `composition` feature https://github.com/RobinTail/express-zod-api/pull/875
I'll take a look on possible fix soon.

**Comment by RobinTail** [comment]
🚀  Fixed in v10.1.3, @shroudedcode
